pycbc/io/hdf.py

```diff
@@ -141,6 +142,7 @@ def __init__(self, data=None, files=None, groups=None):
 
         for k in self.data:
             setattr(self, k, self.data[k])
+        self.groups=groups
```

Comment:
could this be done right at the beginning of the init method just after self.data = data?  atm it's not totally clear if 'groups' might be changed by the block of code in between (I don't think it is)